File: Algoritmos de Ordenação/Python/bubble_sort.py
def bubble_sort(lista):
    for i in range(len(lista)-1):
        trocou = False
        
        for j in range(len(lista)-1-i):
            if lista[j] > lista[j+1]:
                lista[j], lista[j+1] = lista[j+1], lista[j]
                trocou = True

        if not trocou:
            break
    
    return

if __name__ == "__main__":
    from random import randint
    
    lista = [randint(-500,500) for i in range(15)]
    print(lista)
    
    bubble_sort(lista)
    print(lista)

File: Algoritmos de Ordenação/C++/bubble_sort.cpp
#include <random>
#include "extra_functions.h"

void bubble_sort(int* array, int tamanho){
    int i,j;
    bool trocou = false;

    for (i=0; i<tamanho-1; i++){

        for (j=0; j<tamanho-1-i; j++){
            if (*(array+j) > *(array+j+1)){

                *(array+j) += *(array+j+1);
                *(array+j+1) = *(array+j) - *(array+j+1);
                *(array+j) -= *(array+j+1);

                trocou = true;
            }
        }

        if (!trocou)
            break;
    }

    return;
}

int main(){

    // Criação do Array de tamanho Fixo
    int tamanho = 15;
    int* array = new int[tamanho];

    // Preenchimento do array com números inteiros
    preencherArray(array, tamanho);

    printArray(array, tamanho);
    bubble_sort(array, tamanho);
    printArray(array, tamanho);

    delete[] array;

    return 0;
}